NPM_CHANGELOG.md

```diff
@@ -1,5 +1,10 @@
 # NPM Changelog
 
+## [4.0.0]
+
+- Updates AG Grid to 31.1.x
+- AG Grid dependencies now via modules (https://www.ag-grid.com/javascript-data-grid/modules/)
```

Comment:
We should provide a little more detailed information about what users need to do here.

For example, that they need to install the core packages, what additional packages they might install, and how they need to register the modules.

Additionally, we should also add some of this information to the README file

Reply:
I extended the NPM_CHANGELOG at bit and fixed packages names in the repo's README. I don't think we need much more as AG Grid is quite helpful in the browser console:

![screenshot-2024-03-05T13:48:17+01:00](https://github.com/mercurymedia/elm-ag-grid/assets/693447/52d8a461-1907-49dd-abe4-e1cc2b03fe86)

Reply:
Also our examples show how it is wired together anyway.

Reply:
For example, I would be happy so see the following upgrade instructions in the changelog:

- replace `ag-grid-community` dependency with `@ag-grid-community/core`
- replace `ag-grid-enterprise` dependency with `@ag-grid-enterprise/core`
- add module-based dependencies as needed 

```json
    "@ag-grid-community/styles": "^31.1.0",
    "@ag-grid-community/client-side-row-model": "^31.1.0",
    "@ag-grid-enterprise/column-tool-panel": "^31.1.0",
    "@ag-grid-enterprise/filter-tool-panel": "^31.1.0",
    "@ag-grid-enterprise/menu": "^31.1.0",
    "@ag-grid-enterprise/range-selection": "^31.1.0",
    "@ag-grid-enterprise/rich-select": "^31.1.0",
    "@ag-grid-enterprise/row-grouping": "^31.1.0",
    "@ag-grid-enterprise/side-bar": "^31.1.0",
```

- register AgGrid components

```js
import { ModuleRegistry } from '@ag-grid-community/core'
import { ClientSideRowModelModule } from '@ag-grid-community/client-side-row-model'
import { LicenseManager } from '@ag-grid-enterprise/core'
import { ColumnsToolPanelModule } from '@ag-grid-enterprise/column-tool-panel'
import { FiltersToolPanelModule } from '@ag-grid-enterprise/filter-tool-panel'
import { MenuModule } from '@ag-grid-enterprise/menu'
import { RangeSelectionModule } from '@ag-grid-enterprise/range-selection'
import { RichSelectModule } from '@ag-grid-enterprise/rich-select'
import { RowGroupingModule } from '@ag-grid-enterprise/row-grouping'
import { SideBarModule } from '@ag-grid-enterprise/side-bar'

ModuleRegistry.registerModules([
  ClientSideRowModelModule,
  ColumnsToolPanelModule,
  FiltersToolPanelModule,
  MenuModule,
  RangeSelectionModule,
  RichSelectModule,
  RowGroupingModule,
  SideBarModule
]);
```

Reply:
And having some of this information in the README would make it much easier for new users to install the package. The README is displayed on the ELM package page, but the examples are not.

Reply:
This information is comparable to the small section about AgGrid Enterprise. Enterprise is not mandatory, but if users want to use AgGrid Enterprise, it's helpful that we mention it already and provide some guidance. 

It is likely that users will use additional AgGrid modules, and not necessarily all of them know about AgGrid. Yes, they could look it up on the website or take the examples as references, but that's not always that straightforward. Therefore, I think it would be helpful to include some reference about this topic in our README file. It doesn't have to be very detailed. A link to the available AgGrid modules and an example of how to register are probably already enough.

And btw, this is basically the same reason we mention Themes and Polyfills. Convenience.

Reply:
done.